libp2p/transports/transport.nim

```diff
@@ -22,13 +24,15 @@ type
     connections*: seq[Connection]
     handler*: ConnHandler
     multicodec*: MultiCodec
+    serverFlags*: set[ServerFlags]
```

Comment:
`TransportFlags` should be defined as a type and we can't assume that it will always be based on chronos specific flags.